tokio/src/net/tcp/socket.rs

```diff
@@ -125,6 +125,7 @@ impl TcpSocket {
             socket2::Type::STREAM,
             Some(socket2::Protocol::TCP),
         )?;
+        inner.set_nonblocking(true)?;
```

Comment:
There *should* be a way to do this as part of the `Socket::new` initialization. If there isn't, we need to fix that.

Reply:
Socket2 supports [`Type::nonblocking`](https://docs.rs/socket2/latest/socket2/struct.Type.html#method.nonblocking) for this.

Reply:
addressed in 08240cc.